docs/utilities/recipes.md

```diff
@@ -0,0 +1,213 @@
+
+### Special behaviour of the `data` tag
+It is strongly advised to use the `data` tag to define the metadata of your items and blocks. Any item which uses `setHasSubtypes(true)` requires the use of the data field. When it is not used within the ingredients or keys, it will mean any metadata of this item will be accepted, for example: Not defining the data of a sword means even a half broken sword will be accepted for the crafting recipe!
+
+Patterns
+--------
+A pattern will be defined with the `pattern` list. Each string represents one row in the crafting grid and each placeholder character within the String represents a column. As seen in the example above a space means that no item needs to be inserted at that position.
+
+Keys
+----
+A key set is used in combination with patterns and contains keys whose name is the same as the placeholder character in the pattern list which it represents. One key may be defined to represent multiply items as it is the case for the wooden button. This means that the player can use one of the defined items for the crafting recipe, for example different types of wood.
+
+```json
+  "key": {
+     "#": [
+      {
+        "item": "minecraft:planks",
+         "data": 0
+      },
+      {
+        "item": "minecraft:planks",
+        "data": 1
+      }
+    ]
+  }
+```
+
+
+Results
+-------
+Every `recipe` has to have a result tag to define the output item.
+
+When crafting something, you can get out more than one item. This is achieved by defining the `count` number. If this is left out, meaning it doesn't exist within the result block, it defaults to 1. Negative values are not allowed here as an Itemstack cannot be smaller than 0. There is no option to use the `count` number anywhere else than for the result.
+The `data` field is a optional and used to define the metadata of a block or item. It defaults to 0 when it doesn't exist.
+
+!!! note
+
+    Any item which uses `setHasSubtypes(true)` requires the data field. In this case, it is not optional!
+    
+Factories
+---------
+When programing, we speak of a factory as beeing an object to create another object. For more information, you may want to check [Wikipedia][Wikipedia]. To create your own factory, you have to create a `_factories.json`. Within this file a type has to be defined, for example: `recipes`,  `ingredients` or `conditions`. The basic rule is that each type will point to a class reference for example `IRecipeFactory `, `IIngredientFactory` or `IConditionFactory` with a known constructor and input method. At last a `name` which can be later used in your recipes and the fully qualified classname have to be specified which is a class you have to create which implements for example `IRecipeFactory` for a factory of the type `recipes`.
+
+```json
+{
+    "<type>": {
+        "<name>": "<fully qualified classname for the specified type>
```

Comment:
Does the name need to be lowercase?
A quote (") is missing on this line.

Reply:
I don't think the name has to be lowercase. Atleast I never found anyone stating that it has to be lowercase.